### ISSUE TITLE
Unnecessary type field in parameter body schema

### ISSUE THREAD

**Issue by wingsofovnia** [swaggo/swag#525]
**Describe the bug**
Docs doesn't pass Swagger validation because of `"type": "object"` in schema object of a parameter schema together with a "$ref". When a ref exists, the type should be excluded. E.g.
```
parameters:
  - name: body
    in: body
    required: true
    schema:
      type: object ## <-----
      $ref: '#/definitions/MyObj'
```

**Expected behavior**
A valid swagger 2.0 output.

**Your swag version**
master (b09d3f)

**Your go version**
e.g. 1.12.9

**Related to #397**

**Comment by ubogdan** [comment]
Please provide a code example so we can test. 

This issue was solved here
fix issue #397: Unnecessary type field on response objects #459

**Comment by wingsofovnia** [comment]
@ubogdan it was fixed for **response** objects not parameters
```golang
// AddObjects godoc
// @Summary Adds a new object
// @ID addObjects
// @Tags objects
// @Accept json
// @Produce json
// @Param body body main.Object true "..." <<<<<<<<<<<<<<<
// @Success 201 {object} main.Object "..."
// @Failure 400 {object} main.ErrorResponse "Bad Request"
// @Failure 500 {object} main.ErrorResponse "Internal server error, please contact support"
// @Router /objects [post]
func (srv *Server) handleAddObject(w http.ResponseWriter, r *http.Request) {}
```
Generated json:
```
....
   "/objects":{
      "post":{
         "consumes":[
            "application/json"
         ],
         "produces":[
            "application/json"
         ],
         "tags":[
            "objects"
         ],
         "summary":"Adds a new object",
         "operationId":"addObject",
         "parameters":[
            {
               "description":"...",
               "name":"body",
               "in":"body",
               "required":true,
               "schema":{
                  "type":"object",  <<<<<<<<<<<<<<<<<<<<<<<<
                  "$ref":"#/definitions/main.Object"
               }
            }
         ],
         "responses":{
            "201":{
               "description":"......",
               "schema":{
                  "$ref":"#/definitions/main.Object"
               }
            },
            "400":{
               "description":"Bad Request",
               "schema":{
                  "$ref":"#/definitions/main.ErrorResponse"
               }
            },
            "500":{
               "description":"Internal server error, please contact support",
               "schema":{
                  "$ref":"#/definitions/main.ErrorResponse"
               }
            }
         }
      }
   }
....
```


Swagger Validator says:
```
"attribute paths.'/objects'(post).[body].type is unexpected",
```

**Comment by easonlin404** [comment]
fixed by #542 . please reopen it if you still have questions.
